gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/transform/SuppressibleTreePathScannerClassVisitor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.suppressibleerrorprone.transform;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+/**
+ * Forces {@code SuppressibleTreePathScanner} to respect the ignore suppressions option.
+ *
+ * <p>{@code RemoveUnusedSuppressions} must see violations that are normally hidden by suppressions to
+ * determine which suppressions are actually needed. While ErrorProneOptions has an
+ * {@code ignoreSuppressionAnnotations} flag for this purpose, many BugCheckers use
+ * {@code SuppressibleTreePathScanner}, which doesn't respect this flag.
+ *
+ * <p>This class uses bytecode manipulation to patch {@code SuppressibleTreePathScanner::isSuppressed}
+ * to respect the ErrorProneOptions setting.
+ */
+final class SuppressibleTreePathScannerClassVisitor extends ClassVisitor {
```

Comment:
As discussed in person, we should try to get the issue fixed in error-prone before resorting to further asm modification.